genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -0,0 +1,3394 @@
+            if nsimplex == 2:
+                polytope_flag = self.func_epa_init_polytope_2d(i_ga, i_gb, i_b)
+            elif nsimplex == 4:
+                polytope_flag = self.func_epa_init_polytope_4d(i_ga, i_gb, i_b)
+
+            # Polytope 3D could be used as a fallback for 2D and 4D cases, but it is not necessary
+            if nsimplex == 3 or (polytope_flag == EPA_P2_FALLBACK3 or polytope_flag == EPA_P4_FALLBACK3):
+                polytope_flag = self.func_epa_init_polytope_3d(i_ga, i_gb, i_b)
+
+            # Run EPA from the polytope
+            if polytope_flag == 0:
+                i_f = self.func_epa(i_ga, i_gb, i_b)
+                distance = self.distance[i_b]
+
+                if num_max_contacts > 1 and i_f != -1:
+                    # Detect multiple contacts
+                    self.func_multi_contact(i_ga, i_gb, i_b, i_f)
+
+    """
+    GJK algorithms
+    """
+
+    @ti.func
+    def func_gjk(self, i_ga, i_gb, i_b, shrink_sphere):
+        """
+        GJK algorithm to compute the minimum distance between two convex objects.
+        """
+
+        # Simplex index
+        n = 0
+
+        # Final number of simplex vertices
+        nsimplex = 0
+
+        # Number of witness points and distance
+        nx = 0
+        dist = 0.0
+
+        # Lambda for barycentric coordinates
+        _lambda = gs.ti_vec4(1.0, 0.0, 0.0, 0.0)
+
+        # Final return flag
+        return_flag = 0
+
+        # Set initial guess of support vector using the positions.
+        # The support vector should be a non-zero vector.
+        approx_witness_point_obj1 = self._solver.geoms_state[i_ga, i_b].pos
+        approx_witness_point_obj2 = self._solver.geoms_state[i_gb, i_b].pos
+        support_vector = approx_witness_point_obj1 - approx_witness_point_obj2
+
+        if support_vector.dot(support_vector) < self.FLOAT_MIN_SQ:
+            approx_witness_point_obj1.x = approx_witness_point_obj1.x + 0.01
+            approx_witness_point_obj2.x = approx_witness_point_obj2.x - 0.01
+            support_vector = approx_witness_point_obj1 - approx_witness_point_obj2
+
+        support_vector_norm = 0.0
+
+        # Whether or not we need to compute the exact distance.
+        # If we shrink the sphere and capsule, we need to compute the distance.
+        get_dist = shrink_sphere
+
+        # We can use GJK intersection algorithm only for collision detection
+        # if we do not have to compute the distance.
+        backup_gjk = not get_dist
+
+        epsilon = 0.0
+        if not self.func_is_discrete_geoms(i_ga, i_gb, i_b):
+            # If the objects are smooth, finite convergence is not guaranteed,
+            # so we need to set some epsilon to determine convergence
+            epsilon = 0.5 * self.tolerance * self.tolerance
+
+        for i in range(self.gjk_max_iterations):
+            # Compute the current support points
+            support_vector_norm = support_vector.dot(support_vector)
```

Comment:
Just do `support_vector_norm = support_vector.norm()` and compare against `self.FLOAT_MIN`.

You are taking the square root in both cases anyway. 

Or if you like fancy stuffs, you could also do:
```python
support_vector_norm = support_vector.norm_sqr()
dir = -support_vector * ti.rsqrt(support_vector_norm)
```
But this seems overzealous!